File: morph-r2rml-rdb/src/main/scala/es/upm/fi/dia/oeg/morph/r2rml/rdb/engine/IQueryTranslatorFactory.scala
package es.upm.fi.dia.oeg.morph.r2rml.rdb.engine

import es.upm.fi.dia.oeg.obdi.core.model.AbstractMappingDocument
import java.sql.Connection
import es.upm.fi.dia.oeg.obdi.core.engine.IQueryTranslator
import es.upm.fi.dia.oeg.obdi.core.engine.AbstractUnfolder

trait IQueryTranslatorFactory {
	def createQueryTranslator(mappingDocument:AbstractMappingDocument
	    , conn:Connection, unfolder:AbstractUnfolder) : IQueryTranslator;
}

File: odemapster-core/src/main/java/es/upm/fi/dia/oeg/obdi/core/engine/IQueryTranslatorFactory.java
package es.upm.fi.dia.oeg.obdi.core.engine;

import java.sql.Connection;

import es.upm.fi.dia.oeg.obdi.core.model.AbstractMappingDocument;

public interface IQueryTranslatorFactory {
	public IQueryTranslator createQueryTranslator(AbstractMappingDocument mappingDocument
			, Connection conn, AbstractUnfolder unfolder);
}
